packages/gasket-helper-intl/lib/index.d.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * Locale settings and known locale file paths
+ */
+export interface LocaleManifest {
+    /**
+     * Base URL where locale files are served
+     */
+    basePath?: string
+    /**
+     * Path to endpoint with JSON files
+     */
+    localesPath: LocalePathPart
+    /**
+     * Locale to fallback to when loading files
+     */
+    defaultLocale: Locale
+    /**
+     * Mapping of locales to share files
+     */
+    localesMap?: Record<Locale, Locale>
+    /**
+     * Available locale files to content hashes
+     */
+    paths: Record<LocalePath, string>
+}
+
+/**
+ * Partial URL representing a directory containing locale .json files
+ * or a URL template with a `:locale` path param to a .json file.
+ */
+export type LocalePathPart = string;
+/**
+ * URL path to a locale .json file
+ */
+export type LocalePath = string;
+/**
+ * Language code only
+ */
+export type Lang = string;
+/**
+ * Language code with region
+ */
+export type Locale = Lang;
+/**
+ * State of loaded locale files
+ */
+export interface LocalesState {
+    messages: Record<string, string>
+    status: Record<LocalePath, LocaleStatus>
+}
+
+/**
+ * Enum for local status values
+ */
+export type LocaleStatus = string;
+export namespace LocaleStatus {
+    const LOADING: string;
+    const LOADED: string;
+    const ERROR: string;
+}
```

Comment:
This is a bit confusing, and I'm not sure it works correctly. It looks like you're going for more of a TypeScript `enum`:

```suggestion
export enum LocaleStatus {
  LOADING = 'LOADING',
  LOADED = 'LOADED',
  ERROR = 'ERROR'
}
```